climakitae/util/utils.py

```diff
@@ -772,90 +772,102 @@ def summary_table(data: xr.Dataset) -> pd.DataFrame:
     return df
 
 
-def convert_to_local_time(data: xr.DataArray, selections) -> xr.DataArray:
+def convert_to_local_time(data: xr.DataArray | xr.Dataset) -> xr.DataArray:
     """
     Convert time dimension from UTC to local time for the grid or station.
 
     Args:
-        data (xarray.DataArray): Input data.
-        selections: DataParameters object containing selection details.
+        data (xarray.DataArray | xr.Dataset): Input data.
 
     Returns:
         xarray.DataArray: Data with converted time coordinate.
     """
 
+    # Only converting hourly data
+    if "frequency" not in data.attrs:
+        # Make a guess at frequency
+        timestep = pd.Timedelta(
+            data.time[1].item() - data.time[0].item()
+        ).total_seconds()
+        match timestep:
+            case 3600:
+                frequency = "hourly"
+            case 86400:
+                frequency = "daily"
+            case _ if timestep > 86400:
+                frequency = "monthly"
+    else:
+        frequency = str(data.attrs["frequency"])
+
     # If timescale is not hourly, no need to convert
-    if selections.timescale in ["monthly", "daily"]:
+    if frequency != "hourly":
         print(
             "You've selected a timescale that doesn't require any timezone shifting, due to its timescale not being granular enough (hourly). Please pass in more granular level data if you want to adjust its local timezone."
         )
         return data
 
-    # 1. Get the time slice from selections
-    start, end = selections.time_slice
+    # Find out if Stations or Gridded type
+    if "data_type" not in data.attrs:
+        if isinstance(data, xr.core.dataarray.DataArray):
+            print(
+                "Data Array attribute 'data_type' not found. Please set 'data_type' to 'Stations' or 'Gridded'."
+            )
+            return data
+        else:
+            try:
+                # Grab from one of data arrays in dataset
+                variable = list(data.keys())[0]
+                data_type = data[variable].attrs["data_type"]
+            except KeyError:
+                print(
+                    f"Could not find attribute 'data_type' attribute set in {variable} attributes. Please set `data_type` attribute."
+                )
+                return data
+    else:
```

Comment:
if you use the walrus operator this "else" chain should also be able to be removed